ui/desktop/src/components/OllamaSetup.test.tsx

```diff
@@ -0,0 +1,278 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import { OllamaSetup } from './OllamaSetup';
+import * as ollamaDetection from '../utils/ollamaDetection';
+import * as providerUtils from '../utils/providerUtils';
+import { toastService } from '../toasts';
+
+// Mock dependencies
+vi.mock('../utils/ollamaDetection');
+vi.mock('../utils/providerUtils');
+vi.mock('../toasts');
+
+// Mock useConfig hook
+const mockUpsert = vi.fn();
+const mockAddExtension = vi.fn();
+const mockGetExtensions = vi.fn();
+
+vi.mock('./ConfigContext', () => ({
+  useConfig: () => ({
+    upsert: mockUpsert,
+    addExtension: mockAddExtension,
+    getExtensions: mockGetExtensions,
+  }),
+}));
+
+describe('OllamaSetup', () => {
+  const mockOnSuccess = vi.fn();
+  const mockOnCancel = vi.fn();
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    // Default mocks
+    vi.mocked(ollamaDetection.getPreferredModel).mockReturnValue('gpt-oss:20b');
+    vi.mocked(ollamaDetection.getOllamaDownloadUrl).mockReturnValue('https://ollama.com/download');
+  });
+
+  describe('when Ollama is not detected', () => {
```

Comment:
not really sure of the value of this test - inspired by @zanesq, this mocks out a lot, so really just checks the component does something (can remove it if it seems contrived)